towerlib/towerlib.py

```diff
@@ -2018,3 +2034,300 @@ def get_schedule_by_name(self, name):
 
         """
         return next(self.schedules.filter({'name__iexact': name}), None)
+
+    def update_all_organization_projects(self, organization_name):
+        """Update all the projects in ansible tower for a given organization.
+        """
+        organization = self.get_organization_by_name(organization_name)
+        for project in organization.projects:
+            project.update
+
+    def update_project_by_id(self, project_id):
+        """Update the ansible tower project with given project id.
+
+        Args:
+            project_id: The id of the project, which is to be updated.
+
+        Returns:
+            list: List of response of api request as json on success, None otherwise.
+
+        """
+        project = self.get_project_by_id(project_id)
+        return project.update
+
+    def update_organization_project_by_name(self, organization, project_name):
+        """Update the ansible tower project with given project name.
+
+        Args:
+            organization: The name of the organization.
+            project_name: The name of the project, which is to be updated.
+
+        Returns:
+            dict: dict of response of api request as json on success, None otherwise.
+
+        """
+
+        project = self.get_organization_project_by_name(organization, project_name)
+        return project.update
+
+    def update_organization_projects_by_scm_url(self, scm_url, organization_name):
+        """Send update request to update project for a given git repository (scm_url) withing an organization.
+
+        Args:
+            scm_url: the http url of the required repository.
+
+        """
+        organization = self.get_organization_by_name(organization_name)
+        matching_projects = [project for project in organization.projects if project.scm_url == scm_url]
+        for project in matching_projects:
+            self._logger.debug("A request is being sent to update the project with the name '{}' and with scm url '{}'"
+                               .format(project.name, scm_url))
+            project.update
+
+    def update_organization_projects_by_branch_name(self, scm_url, branch_name, organization_name):
+        """Update an ansible tower project or list of projects for an organization based on their branch name.
+
+        A scm_branch can only be identified correctly with a corresponding scm_url.
+
+        Args:
+            scm_url: the URL of the relevant repository configured in the project.
+            branch_name: the name of the branch, which is selected as scm_branch parameter of the project.
+
+        """
+        organization = self.get_organization_by_name(organization_name)
```

Comment:
All the above comments apply. Make safe in case of no org, make generator comprehension for responsiveness and call the method, don't just refer to it.